src/renovate.ts

```diff
@@ -113,6 +117,27 @@ class Renovate {
       );
     }
   }
+
+  private validateDockerCmdFileArgument(): void {
+    const dockerCmdFile = this.input.getDockerCmdFile();
+    if (dockerCmdFile !== null) {
+      if (
+        !fs.existsSync(dockerCmdFile) ||
+        !fs.statSync(dockerCmdFile).isFile()
+      ) {
+        throw new Error(
+          `dockerCmdFile '${dockerCmdFile}' MUST be an existing file`,
+        );
+      }
+      try {
+        fs.accessSync(dockerCmdFile, fs.constants.R_OK | fs.constants.X_OK);
+      } catch {
+        throw new Error(
+          `dockerCmdFile '${dockerCmdFile}' MUST have read and execute rights`,
+        );
+      }
```

Comment:
only use one stat call for this, it gives all information you need.

please also use async `fs/promises` if possible.

Reply:
thanks, will change that